### PR TITLE
Added new permeability material to PorousFlow

### LINKED ISSUE #11611: Add PorousFlow permeability material for variable multiplied by tensor components
## Rationale

It is sometimes desirable to calculate permeability from a scalar multiplied by tensor components, where the scalar might be some function of depth, for example, but the tensor components (representing the anisotropy of the permeability) remain constant. Currently this can be achieved using PorousFlowPermeabilityConstFromVar but this requires a separate variable for each component of the permeability tensor, resulting in untidy and error-prone input files.

## Description

Add new PorousFlow permeability material enabling user to specify permeability in terms of a coupled var multiplied by tensor components.

## Impact

Ability to specify permeability in terms of 2 input parameters (a scalar variable and a tensor).

### PR DESCRIPTION
**Pull request by hsheldon**
closes #11611

### CODE REVIEW

File: modules/porous_flow/test/tests/poroperm/PermTensorFromVar01.i [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+  nl_rel_tol = 1E-8
+  l_max_its = 200
+  nl_max_its = 400
+
+  petsc_options_iname = '-pc_type -pc_asm_overlap -sub_pc_type -ksp_type -ksp_gmres_restart'
+  petsc_options_value = ' asm      2              lu            gmres     200'
+[]
+
+[Outputs]
+  file_base = PermTensorFromVar01
```
Comment (permcody) [inline modules/porous_flow/test/tests/poroperm/PermTensorFromVar01.i]: Unnecessary parameter. Applies to most (or all) of these new test cases.

Comment (hsheldon) [reply modules/porous_flow/test/tests/poroperm/PermTensorFromVar01.i]: I'm confused. Surely its needs a file name in order to perform CSVDiff?

**Comment by permcody** [comment]
By default, the name is the input file name with “_out” appended and then
the appropriate suffix added. That way if you change the name of the input
file you get a new output file without having to remember editing this
parameter.

**Comment by hsheldon** [comment]
Ok, fixed.

File: modules/porous_flow/src/materials/PorousFlowPermeabilityTensorFromVar.C [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  return params;
+}
+
+PorousFlowPermeabilityTensorFromVar::PorousFlowPermeabilityTensorFromVar(
+    const InputParameters & parameters)
+  : PorousFlowPermeabilityBase(parameters),
+    _perm(coupledValue("perm")),
+    _k_anisotropy(parameters.isParamValid("k_anisotropy")
+                      ? getParam<RealTensorValue>("k_anisotropy")
+                      : RealTensorValue(1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0))
```
Comment (permcody) [inline modules/porous_flow/src/materials/PorousFlowPermeabilityTensorFromVar.C]: Suggestion: This is fine, but you can make this a little more compact by supply the default right in the declaration of the parameter. Then you don't have to have a conditional here where you "get" the value.

**Review by permcody — APPROVED** [review]

**Pull request merged**